addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+		return this.connection;
+	}
+
+	/**
+	 * Updates things from device model.
+	 * 
+	 * @param thing
+	 *            Thing to be updated.
+	 * @param device
+	 *            Device model with new data.
+	 */
+	private void updateThingFromDevice(Thing thing, DeviceModel device) {
+		if (thing == null || device == null) {
+			LOG.error("thing or device null, cannot perform update");
+			return;
+		}
+		LOG.debug("about to update " + thing.getUID() + " from "
+				+ device.toString());
+		if (device.isTempSensor()) {
+			Channel channel = thing.getChannel(CHANNEL_TEMP);
+			this.updateState(channel.getUID(), new DecimalType(device
+					.getTemperature().getCelsius()));
+		}
+		if (device.isPowermeter()) {
+			Channel channelEnergy = thing.getChannel(CHANNEL_ENERGY);
+			this.updateState(channelEnergy.getUID(), new DecimalType(device
+					.getPowermeter().getEnergy()));
+			Channel channelPower = thing.getChannel(CHANNEL_POWER);
+			this.updateState(channelPower.getUID(), new DecimalType(device
+					.getPowermeter().getPower()));
+		}
+		if (device.isSwitchableOutlet()) {
+			Channel channel = thing.getChannel(CHANNEL_SWITCH);
+			if (device.getSwitch().getState().equals(SwitchModel.ON)) {
+				this.updateState(channel.getUID(), OnOffType.ON);
+			} else if (device.getSwitch().getState().equals(SwitchModel.OFF)) {
+				this.updateState(channel.getUID(), OnOffType.OFF);
+			} else {
+				LOG.error("unknown state " + device.getSwitch().getState()
+						+ " for channel " + channel.getUID());
+			}
+		}
+	}
+
+	/**
+	 * Builds a {@link ThingUID} from a device model.
+	 * 
+	 * @param device
+	 * @return
+	 */
+	public ThingUID getThingUID(DeviceModel device) {
+		ThingUID bridgeUID = this.getThing().getUID();
+		ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, device
+				.getProductName().replaceAll("[^a-zA-Z0-9_]", "_"));
+
+		if (FritzAhaHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
+			String thingName = device.getName()
+					.replaceAll("[^a-zA-Z0-9_]", "_");
+			ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, thingName);
+			return thingUID;
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Start the polling.
+	 */
+	private synchronized void onUpdate() {
+		if (this.getThing() != null) {
+			if (pollingJob == null || pollingJob.isCancelled()) {
+				LOG.debug("start polling job at intervall " + refreshInterval);
+				pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 1,
+						refreshInterval, TimeUnit.SECONDS);
+			} else {
+				LOG.warn("pollingJob null or cancelled");
+			}
+		} else {
+			LOG.warn("bridge is null");
+		}
+	}
+	/**
+	 * Just logging - nothing to do.
+	 */
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		LOG.debug("update " + channelUID.getAsString() + " with "
+				+ command.toString());
+	}
+	/**
+	 * Start search for new AHA devices connected to this bridge.
+	 * @param service
+	 */
+	public void startSearch(FritzBoxDiscoveryService service) {
```

Comment:
This method can imho be completely removed here as the code can be directly put into FritzBoxDiscoveryService.startScan()